clients/admin-ui/src/features/taxonomy/hooks.tsx

```diff
@@ -66,7 +78,33 @@ const transformTaxonomyBaseToInitialValues = (t: TaxonomyEntity) => ({
   is_default: t.is_default ?? false,
 });
 
+const transformBaseFormValuesToEntity = (
+  initialValues: FormValues,
+  newValues: FormValues
+) => {
+  const isCreate = initialValues.fides_key === "";
+
+  // parent_key and fides_keys are immutable
+  // parent_key also needs to be undefined, not an empty string, if there is no parent element
+  const entity = { ...newValues };
+  if (isCreate) {
+    const parentKey = parentKeyFromFidesKey(newValues.fides_key);
+    entity.parent_key = parentKey === "" ? undefined : parentKey;
+  } else {
+    entity.parent_key =
+      initialValues.parent_key === "" ? undefined : initialValues.parent_key;
+    entity.fides_key = initialValues.fides_key;
+  }
+
+  delete entity.definitionIdToCustomFieldValue;
```

Comment:
Note that the FormValues type now knows that this property exists (instead of just getting snuck in by Formik).

Reply:
heh nice, I always felt a little too sneaky about that